### PR TITLE
Reduced size of acceptance tests, part 1

### PR DESCRIPTION
**Pull request by tjtg**
Reduce the size of acceptance test data for:
- apply-night-mask
- generate-realizations
- weighted-blending
- wind-downscaling

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

This PR contributes towards #1804

### CODE REVIEW

**Comment by bayliffe** [comment]
### Progress

- I have updated a copy of Tom's acceptance test data branch to merge in changes we've made to weighted-blending (attribute changes). This can be found in the ppdev test data repository named `smaller_acc_test1`.
- I've rebased the code changes on IMPROVER following on the recent changes and updated the checksums for the above branch. This has been pushed to this PR (force pushed, sorry).
- I've started plotting out the data and checking that the reduced domains still test the important things we hoped the tests would cover.

**Comment by bayliffe** [comment]
### Issues

The new test data appears to have been carefully selected to ensure we are capturing the required elements in our tests. There is only one issue that I have come across.

`weighted_blending/spatial_weights/nowcast_data/`
This nowcast data has been trimmed to a region that is entirely covered by the radar domain. This means that when the data is used in blending tests, the use or non-use of a fuzzy interface between the radar and model data has no impact. This can be seen by comparing the data in the new trimmed KGOs, which are now identical with or without using fuzziness.
```
nccmp -ds model.nc model_no_fuzzy.nc 
Files "model.nc" and "model_no_fuzzy.nc" are identical.
nccmp -ds cycle.nc cycle_no_fuzzy.nc 
Files "cycle.nc" and "cycle_no_fuzzy.nc" are identical.
```
An equivalent check of the previous KGOs reveals that we expect to see the impact of the softening at the edge of the radar domain. The chosen domain for these tests needs to capture the edge of the masked region of nowcast data.


### Actions

I've moved the smaller domain for the spatial-weights tests to the edge of the radar domain so that we capture a mix of masked and unmasked regions. The KGO with and without fuzziness no longer contain the same data. The domain is plotted below, it capture the edge of the Netherlands just south of Rotterdam.

Nowcast
nowcast_domain

UKVX
ukvx_domain

KGO model blend with a fuzzy boundary
model_blend

KGO model blend without a fuzzy boundary
model_blend_no_fuzzy

I have added these changes as a commit to the acceptance test data and pushed up another commit to this PR.


### Notes

The acceptance test data appears to contain a lot of changes (when performing a `git diff --name-only`) that are not related to the listed CLIs for which the domains have been changed. These appear to be the restoration of soft links instead of duplicate files.

**Comment by bayliffe** [comment]
@MoseleyS this needs another blue team review please. Fresh eyes would be useful to pick out anything I've missed.

**Comment by gavinevans** [comment]
Please note that I have updated this PR by rebasing on master. This was due to conflicts that had arisen due to modifications to the `generate-realizations` acceptance test data as part of #1838. Note that I haven't modified the acceptance test data added in #1838, which was already using a relatively reduced size domain covering the area to the north of Scotland, as shown below.

generate_realizations_wind_input

**Comment by gavinevans** [comment]
Thanks for the work on this.

I've used the notebook in https://github.com/MetOffice/improver_aux/pull/347 to help with this review. In a few cases, I can see that the data within the acceptance tests data has changed beyond just a reduction in the domain. However, in all cases, these changes are within the known good outputs with the changes to the acceptance test data domain presumably having an impact on the calculations. 

I've also noticed that the files within weighted_blending/realizations and weighted_blending/cycletime have not changed, whilst I would have expected that these files would change. I will create a version of the acceptance test data with updates to the files in these two directories to reduce the domain size.

**Comment by gavinevans** [comment]
I've now pushed up an amendment to the weighted_blending/realizations files. This updated acceptance test data is within my smalleracctest2 branch. @bayliffe, please could you have another look at this PR including my changes?

**Comment by bayliffe** [comment]
## Update

Changes [described here](https://github.com/metoppv/improver/issues/1804#issuecomment-1386924050)

PR now associated with acceptance test data branch: `shrink_atd_1`

[Notebook plotting comparisons](https://github.com/MetOffice/improver_aux/blob/acceptance_test_data_shrinking/notebooks/acceptance_test_data/shrink_atd_1.ipynb) between the original and modified data.

File: improver_tests/acceptance/test_generate_realizations.py [hunk 1]
```diff
@@ -55,7 +54,6 @@ def test_percentiles(tmp_path):
     acc.compare(output_path, kgo_path)
 
 
-@pytest.mark.slow
 def test_percentiles_reordering(tmp_path):
```
Comment (gavinevans) [inline improver_tests/acceptance/test_generate_realizations.py:57]: There is one instance of `@pytest.mark.slow` in this file that can now be removed, as all tests take less than 1 second.

File: improver_tests/acceptance/test_weighted_blending.py [hunk 2]
```diff
@@ -155,7 +155,6 @@ def test_invalid_nonlin_lin(tmp_path):
         run_cli(args)
 
 
-@pytest.mark.xfail(run=False, reason="takes ~5 minutes to run")
 def test_percentile(tmp_path):
```
Comment (gavinevans) [inline improver_tests/acceptance/test_weighted_blending.py:158]: There are a few instances of `@pytest.mark.slow` in this file that can now be removed, as all tests take less than 1 second.

**Review by gavinevans — CHANGES_REQUESTED** [review]
Thanks for the updates, @bayliffe 👍 

The coarsening of the domain helps to reduce data volumes whilst retaining the meteorological context, which is good. As a note to others, in addition to 1) coarsening and 2) reducing the spatial domain, if coarsening the domain is not suitable for the algorithm, the wind downscaling inputs on vertical levels have also been coarsened to 5, 50, 200, 500, 1000, 2000, 3000, 4000, 6000m. 

I've added a couple of very minor comments.

File: improver_tests/acceptance/test_weighted_blending.py (hunk 2 above)
Comment (bayliffe) [reply improver_tests/acceptance/test_weighted_blending.py:156]: Thanks Gavin, good spot.

**Review by gavinevans — APPROVED** [review]

**Pull request merged**